commands/command_edit.cpp

```diff
@@ -1285,6 +1286,58 @@ void EditCylinder::undo()
 	redo();
 }
 
+EditSensors::EditSensors(int cylIndex, divecomputer *dcIn, std::vector<int> sensors)
+	: EditDivesBase(true), dc(dcIn)
+{
+	if (dives.empty())
+		return;
+
+	setText(Command::Base::tr("Edit sensors"));
+
+	for (int s = 0; s < MAX_SENSORS; ++s) {
+		if (std::find(sensors.begin(), sensors.end(), s) != sensors.end()) {
+			// If this sensor is in the list of sensors being added to this cylinder
+			// then store the old cylinder in undoMapping and the new cylinder in redoMapping.
+			undoMapping.emplace_back(s, dc->sample[0].sensor[s]);
+			redoMapping.emplace_back(s, cylIndex);
+		} else if (dc->sample[0].sensor[s] == cylIndex) {
+			// Otherwise, if the sensor was previously attached to this cylinder then
+			// save that in the undoMapping and set it to NO_SENSOR in redoMapping so
+			// it is not attached to any cylinder.
+			undoMapping.emplace_back(s, cylIndex);
+			redoMapping.emplace_back(s, NO_SENSOR);
+		}
+	}
+
+	mapSensors(redoMapping);
```

Comment:
Not here, please. Any action is performed in redo()/undo().

Reply:
Ah, so that's how the redo()/undo() system works.